src/Containers/Microsoft.NET.Build.Containers/ContainerHelpers.cs

```diff
@@ -281,4 +282,86 @@ internal static bool IsValidEnvironmentVariable(string envVar)
             return false;
         }
     }
+
+#if NET8_0_OR_GREATER
+
+    /// <summary>
+    /// The APP_UID environment variable is a convention used to set the user in a data-driven manner. we should respect it if it's present.
+    /// </summary>
+    internal static void AssignUserFromEnvironment(ImageBuilder imageBuilder, ILogger logger)
+    {
+        // it's a common convention to apply custom users with the APP_UID convention - we check and apply that here
+        if (imageBuilder.EnvironmentVariables.TryGetValue(KnownStrings.EnvironmentVariables.APP_UID, out string? appUid))
+        {
+            logger.LogTrace("Setting user from APP_UID environment variable");
+            imageBuilder.SetUser(appUid);
+        }
+    }
+
+    internal static string[] Split(string input)
+    {
+        return input.Split(';', StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+    }
+
+    /// <summary>
+    /// This is a parser for ASPNETCORE_URLS based on https://github.com/dotnet/aspnetcore/blob/main/src/Http/Http/src/BindingAddress.cs
+    /// We can cut corners a bit here because we really only care about ports, if they exist.
+    /// </summary>
+    internal static Regex aspnetPortRegex = new Regex(@"(?<scheme>\w+)://(?<domain>([*+]|).+):(?<port>\d+)");
+
+    /// <summary>
+    /// ASP.NET can have urls/ports set via three environment variables - if we see any of them we should create ExposedPorts for them
+    /// to ensure tooling can automatically create port mappings.
+    /// </summary>
+    internal static void AssignPortsFromEnvironment(ImageBuilder imageBuilder, ILogger logger)
+    {
+        // asp.net images control port bindings via three environment variables. we should check for those variables and ensure that ports are created for them
+        if (imageBuilder.EnvironmentVariables.TryGetValue(KnownStrings.EnvironmentVariables.ASPNETCORE_HTTP_PORTS, out string? httpPorts))
+        {
+            logger.LogTrace("Setting ports from ASPNETCORE_HTTP_PORTS environment variable");
```

Comment:
as discussed, it would be good to document this logic and env var used in the reference.